kiosk/src/Components/EnterHighScore.tsx

```diff
@@ -44,110 +100,56 @@ const EnterHighScore: React.FC<IProps> = ({ kiosk }) => {
     const renderInitials = (): JSX.Element[] => {
         const elements = [];
 
-        console.log("Initial index: " + initialIndexRef.value);
-
         for (let lcv = 0; lcv < configData.HighScoreInitialsLength; lcv++) {
             const thisIndex = lcv;
             elements.push(
-                <HighScoreInitial kiosk={kiosk} key={lcv} isSelected={thisIndex === initialIndex}
+                <HighScoreInitial kiosk={kiosk} key={lcv} isSelected={thisIndex === timesAPressed}
                     onCharacterChanged={character => updateInitial(thisIndex, character[0])} />
             );
         }
 
         return elements;
     }
 
-    const previousInitial = () => {
-        initialIndexRef.value = (initialIndexRef.value + configData.HighScoreInitialsLength - 1) % configData.HighScoreInitialsLength;
-        setInitialIndex(initialIndexRef.value);
-    }
-
-    const nextInitial = () => {
-        initialIndexRef.value = (initialIndexRef.value + 1) % configData.HighScoreInitialsLength;
-        setInitialIndex(initialIndexRef.value);
+    const renderList = (highScores: HighScore[]): JSX.Element[] => {
```

Comment:
It's nice to make helpers like these into functional components -- pretty easy and makes the end component look 'prettier' / more html like. basically the only thing you need to do is swap over so it takes props containing a high score instead of just high score directly, here's a quick example of a similar change https://github.com/microsoft/pxt/pull/9231/commits/b1a921014606db8ec74b984a590b6352ab25d2c3